testing/solomachine.go

```diff
@@ -13,10 +13,18 @@ import (
 	"github.com/stretchr/testify/require"
 
 	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	connectiontypes "github.com/cosmos/ibc-go/v6/modules/core/03-connection/types"
 	commitmenttypes "github.com/cosmos/ibc-go/v6/modules/core/23-commitment/types"
 	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
 	"github.com/cosmos/ibc-go/v6/modules/core/exported"
 	solomachine "github.com/cosmos/ibc-go/v6/modules/light-clients/06-solomachine"
+	ibctm "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint"
+)
+
+var (
+	clientIDSolomachine     = "client-on-solomachine"     // clientID generated on solo machine side
```

Comment:
nit: should we use `06-solomachine-0`? so it follows convention?

Reply:
No, that identifier will be used on the tendermint chain (chainA). I would find it very hard to debug if both sides had the same identifier, I find it easier to work through the code with unique identifiers for each side. Does that make sense?

Reply:
Ah okay, I see your reasoning! So chainA would be a regular tm chain and the counterparty would be a solomachine implementation with a completely different ID. I agree for debugging its better to keep like this.

In practice, it would need to follow the standard client ID structure to pass [`ValidateClientType`](https://github.com/cosmos/ibc-go/blob/1b7745f30ff1e35640d121d449087aa5b8910793/modules/core/02-client/types/client.go#L90) tho, right? That is for the solomachine client on the tm chain. 
Doesn't that mean that the solomachine(chainB) must have a `07-tendermint` client for tracking its counterparty (chainA)?

Reply:
For the solomachine on the tm chain that is correct. The identifier in question is the tendermint client on the solomachine though. 

> Doesn't that mean that the solomachine(chainB) must have a 07-tendermint client for tracking its counterparty (chainA)?

Correct, but we just mock this behaviour (pretend it exists). We convince chainA exists in the connection handshake (generation of the tm client state/consensus state)